### ISSUE TITLE
Прикрутить рабочий код для подсчета MI к реализации алгоритма оптимизации gradient descent с целью улучшения его работы

### ISSUE THREAD

**Issue by Huma97** [Huma97/Information-Bottleneck-Theory-of-Deep-Learning#3]
1) Ознакомиться с реализацией gradient descent на pytorch
2) Найти способ на каждой итерации считать MI слоев и обновлять веса, гарантируя при этом наибольший рост MI каждой последующей итерации
3) Убедиться в лучшей сходимости метода для простых нейронок на датасете вроде MNIST или CIFAR10
